specutils/fitting/fitmodels.py

```diff
@@ -1,20 +1,214 @@
 from __future__ import division
 
 import operator
+import itertools
 
 import numpy as np
+from scipy.signal import convolve
 import astropy.units as u
-from astropy.modeling import optimizers
+from astropy.stats import sigma_clipped_stats
 
 from ..manipulation.utils import excise_regions
+from ..analysis import fwhm, centroid
 from ..utils import QuantityModel
 from ..manipulation import extract_region
 from ..spectra.spectral_region import SpectralRegion
 from ..spectra.spectrum1d import Spectrum1D
 from astropy.modeling import fitting, Model, models
+from astropy.table import QTable
+
+
+__all__ = ['find_lines_threshold', 'find_lines_derivative', 'fit_lines', 'estimate_parameters']
+
+# Define the initial estimators
+#   This is the default methods to use to estimate astropy model
+#   parameters. This is based on only a small subset of the astropy
+#   models but it was determined that this is a decent start as most
+#   fitting will probably use one of these.
+#
+#   Each method list must take a Spectrum1D object and should return
+#   a Quantity.
+
+parameter_estimators = {
+    'Gaussian1D': {
+        'amplitude': lambda s: max(s.flux),
+        'mean': lambda s: centroid(s, region=None),
+        'stddev': lambda s: fwhm(s) 
+    },
+    'Lorentzian1D': {
+        'amplitude': None,
+        'x_0': None,
+        'fwhm':None 
+    },
+    'Voigt1D': {
+        'x_0': None,
+        'amplitude_L': None,
+        'fwhm_L': None,
+        'fwhm_G': None 
+    },
+    'Const1D': {
```

Comment:
Hmmm... maybe I am confused then. If the spectrum has a DC offset (for whatever reason) and one wants to fit a line on top of the DC offset, can one specify an offset in the Gaussian1D (for example) model? I didn't see something like that. If not, then how does one specify a fit like this?

Reply:
Ah,  what I imagine as the flow is:

1. Get the continuum model
2. subtract the continuum
3. run the parameter estimators
4a. fit the lines to the subtracted spectrum
OR
4b .fit the lines+continuum to the original spectrum

But because the analysis tools operate on sub/normed spectrum, I think it's more consistent to assume the same for the line-estimator. 

We may want to later *add* the combined-estimator, but I think that's an add-on not v1